src/python/gudhi/tensorflow/_layer.py

```diff
@@ -0,0 +1,16 @@
+import tensorflow
+# The sklearn parse_version function is taken from 'packaging' and this dependency is not mandatory
+from sklearn.utils.fixes import parse_version
```

Comment:
I am not sure what the comment means.
Do we now need to install sklearn to be able to use the tensorflow stuff?
Would `tuple(map(int, tf.__version__.split('.')[:2])) < (2,16)` be good enough?